test/cppsuite/test_harness/workload/database_operation.h

```diff
@@ -165,35 +167,45 @@ class database_operation {
             cursors.push_back(cursor);
         }
 
-        cpt = 0;
-        /* Walk each cursor. */
-        for (const auto &it : cursors) {
-            collection_name = collection_names[cpt];
-            /* Walk each key. */
-            for (keys_iterator_t iter_key = context.get_collection_keys_begin(collection_name);
-                 iter_key != context.get_collection_keys_end(collection_name); ++iter_key) {
-                /* Do not process removed keys. */
-                if (!iter_key->second.exists)
-                    continue;
-
+        /*
+         * Update each collection while the test is running.
+         */
+        i = 0;
+        while (context.is_running() && !collection_names.empty()) {
+            if (i >= collection_names.size())
+                i = 0;
+            ret = cursors[i]->next(cursors[i]);
+            /* If we have reached the end of the collection, reset. */
+            if (ret == WT_NOTFOUND) {
+                cursors[i]->reset(cursors[i]);
+                ++i;
+                continue;
+            }
+            /* Stop updating in case of an error. */
+            else if (ret != 0) {
+                error_message = "update_operation: cursor->next() failed " + std::to_string(ret);
+                debug_print(error_message, DEBUG_ERROR);
+                break;
+            } else {
+                testutil_check(cursors[i]->get_key(cursors[i], &key));
+                generated_value =
+                  random_generator::random_generator::instance().generate_string(value_size);
                 ts = context.get_timestamp_manager()->get_next_ts();
 
                 /* Start a transaction if possible. */
                 if (!context.is_in_transaction()) {
                     context.begin_transaction(session, "");
                     context.set_commit_timestamp(session, ts);
                 }
-                generated_value =
-                  random_generator::random_generator::instance().generate_string(value_size);
-                testutil_check(update(context.get_tracking(), it, collection_name,
-                  iter_key->first.c_str(), generated_value.c_str(), ts));
+
+                testutil_check(update(context.get_tracking(), cursors[i], collection_names[i], key,
```

Comment:
Can we change the convention to having update() and similar calls do error checking internally? I think that makes more sense rather than expecting the user to know to wrap everything in testutil_check. What do you think?

Reply:
I agree, it makes the user experience better too. The impacted functions are `insert` and `update`.